File: analyzer/testdata/analyzer/java/const-remove/base/remove.java
public class remove {

    int v1 = 0;

    private static final int c1 = 42;

    int v2 = 1;
    
    private final int c2 = 7;

    int v3 = 2;

    private final int c3 = c1 + 7;

    int removeLocalA() {
        int tmp1 = 0;
        final int c4 = 42;
        int tmp2 = 0;
        return tmp1 + tmp2;
    }
    
    int removeLocalB(boolean p) {
        final int c4 = 42;
        int tmp1 = 0;
        if (p) {
            int tmp2 = 1;
            final int c5 = 7;
        }        
        return tmp1;
    }
}

File: analyzer/testdata/analyzer/go/const-remove/base/remove.go
package remove

// removeLocalA tests removing of locally defined constants (variant a)
func removeLocalA() int {

	v1 := 0
	var v2 int

	const c = 42

	v3 := 0

	return v1 + v2 + v3
}

// removeLocalB tests removing of locally defined constants (variant b)
func removeLocalB(b bool) int {

	const c1 = 42

	v := 0

	if b {
		v = 1
		const c2 = 7
	}

	return v
}
